bindings/java/java/src/main/java/com/dioxuslabs/taffy/style/Style.java

```diff
@@ -0,0 +1,237 @@
+package com.dioxuslabs.taffy.style;
+
+import com.dioxuslabs.taffy.enums.*;
+import com.dioxuslabs.taffy.geom.Line;
+import com.dioxuslabs.taffy.geom.Point;
+import com.dioxuslabs.taffy.geom.Rect;
+import com.dioxuslabs.taffy.geom.Size;
+import com.dioxuslabs.taffy.geom.grid.GenericGridPlacement;
+import com.dioxuslabs.taffy.geom.grid.NonRepeatedTrackSizingFunction;
+import com.dioxuslabs.taffy.geom.grid.TrackSizingFunction;
+import com.dioxuslabs.taffy.geom.measure.Dimension;
+import com.dioxuslabs.taffy.geom.measure.LengthPercentage;
+import com.dioxuslabs.taffy.geom.measure.LengthPercentageAuto;
+
+import java.util.List;
+
+@SuppressWarnings("all")
+public class Style {
+    private Display display;
+    private boolean itemIsTable;
+    private BoxSizing boxSizing;
+
+    private Point<Overflow> overflow;
+    private float scrollbarWidth;
+
+    private Position position;
+    private Rect<LengthPercentageAuto> inset;
```

Comment:
You could consider "flattening" the fields that use `Rect`, `Size`, and `Line` (so have `width` and `height` fields instead of `size`). I reckon that would simplify the API quite a bit (and mean you wouldn't need to define the `geom` java types or conversions at all.

Reply:
I guess I could look at it, but what about situations where only a part of components of a `geom` is defined? (let's say on `left` is defined in a `Rect`)